.github/workflows/build.yml

```diff
@@ -42,6 +42,80 @@ jobs:
             -Psonatype.username=${{ secrets.SONATYPE_USERNAME }}
             -Psonatype.password=${{ secrets.SONATYPE_PASSWORD }}
 
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - uses: gradle/wrapper-validation-action@v1
+      - name: Unit tests
+        run: ./gradlew testDebugUnitTest
+        working-directory: .
+      - name: Upload failed unit test artifacts
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: unit-test-failures
+          path: '**/build/reports'
+
+  instrumentation-tests:
+    name: Instrumentation tests
+    runs-on: macOS-latest
+    timeout-minutes: 20
+    strategy:
+      fail-fast: false
+      matrix:
+        api-level: [31]
```

Comment:
No AVD exists for API 32 without Google APIs

https://gist.github.com/alvr/8db356880447d2c4bbe948ea92d22c23

If we want to use API-level 32 (which is the compile target), we would need to set `target: google_apis`. That would then mean the library could hypothetically not work on stock Android.

Reply:
1. AFAIK 32 is a sub-release with low level of adoption because it is targeted only for foldable devices and tablets.
2. Testing on 31 is fine too.
3. 33 release is [soon](https://android-developers.googleblog.com/2022/07/Final-Android-13-Beta-update-official-release-is-next.html).

So just use 31 (because non-Google images are less bloated.

Reply:
Sounds good! thanks :)